.github/workflows/tests.yml

```diff
@@ -0,0 +1,41 @@
+name: Tests and linting
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install poetry
+        run: pipx install poetry
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+          cache: "poetry"
+      - name: Install dependencies
+        run: poetry install
```

Comment:
I'm confused how this actually works. I mean, the test passes, but since we don't call it with `--with dev`, where does it get `pytest` etc. from?

Reply:
I think they changed this a few times recently. But also locally I never passed any flags like `--with-dev` and I have all the packages installed.

Reply:
ah I got so used to having to do it manually all the time :slightly_smiling_face: